src/MultiIndex.zig

```diff
@@ -729,6 +823,19 @@ pub fn createIndexInternal(
         return error.InvalidIndexName;
     }
 
+    // If restore_from is provided, use restore path
+    if (options.restore_from) |snapshot_url| {
+        // Start restoration using existing restore logic
+        try self.startIndexRestore(index_name, options.generation orelse 1, snapshot_url);
```

Comment:
Don't assume generation is 1, this needs to come from the existing index_redirect, maybe it would be better not to pass it as generation but as IndexOptions.